### PR TITLE
Add support for the GraphQL Admin API

### PR DESCRIPTION
**Pull request by nneufeld**
Support for the new GraphQL Admin API 🎉 

https://help.shopify.com/api/graphql-admin-api

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -221,6 +221,28 @@ gem install shopify_cli
...
+The authentication process (steps 1-5 under [Getting Started](#getting-started))
+is identical. Once your session is activated, simply construct a new graphql
+client and use `parse` and `query` as defined by
+[graphql-client](https://github.com/github/graphql-client#defining-queries).
+
+```ruby
+client = ShopifyAPI::GraphQL.new
+
+SHOP_NAME_QUERY = client.parse <<-'GRAPHQL'
+  shop {
```
Comment (aldrienht) [inline README.md]: @nneufeld getting `GraphQL::ParseError: Parse error on "shop" (IDENTIFIER) at [1, 1]` in `shop{name}`
I tried `{shop{name}}` and works :)

Comment (aldrienht) [inline README.md]: @nneufeld i played this in rails console, and getting GraphQL::ParseError: Parse error on "shop" (IDENTIFIER) at [1, 1] for shop {name} but works with {shop {name}}

And also, I'm having trouble creating a simple rails model/controller method.
I'm getting GraphQL::Client::DynamicQueryError Expected definition to be assigned to a static constant
I tried changing case for variable but no luck. Please help.

File: lib/shopify_api/resources/graphql.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+require 'graphql/client'
+require 'graphql/client/http'
+
+module ShopifyAPI
+  # GraphQL API.
+  class GraphQL
+    def initialize
+      uri = Base.site.dup
+      uri.path = '/admin/api/graphql.json'
+      @client = ::GraphQL::Client::HTTP.new(uri.to_s) do
```
Comment (radar) [inline lib/shopify_api/resources/graphql.rb]: Wouldn't this code also need to pass in the `X-Shopify-Storefront-Access-Token` header? I don't see where that is configured here at all.

Comment (eapache) [reply lib/shopify_api/resources/graphql.rb]: The storefront-access-token header is only necessary for the storefront API (https://help.shopify.com/api/custom-storefronts/storefront-api). This PR is for the admin API.

Comment (lopert) [reply lib/shopify_api/resources/graphql.rb]: Should @client here be @http?

**Comment by yoonwaiyan** [comment]
Hi any update of this gem? We'll eager to try out using GraphQL queries.

File: lib/shopify_api/resources/graphql.rb (hunk 2 above)
Comment (eapache) [reply lib/shopify_api/resources/graphql.rb]: I think so. It doesn't affect the correctness of the code but it would be less confusing.

**Comment by nneufeld** [comment]
@yoonwaiyan Sorry for the delay on this, I'll get a new version of the gem released with this today.

File: README.md [hunk 3]
```diff
@@ -221,6 +221,30 @@ gem install shopify_cli
    shopify-cli help
    ```
 
+## GraphQL
+
+This library also supports Shopify's new [GraphQL API](https://help.shopify.com/api/reference)
```
Comment (lopert) [inline README.md]: This link should link to:
https://help.shopify.com/api/graphql-admin-api

**Comment by jamiemtdwyer** [comment]
🙌

**Pull request merged**